### ISSUE TITLE
Reusing http::client::response

### ISSUE THREAD

**Issue by tpecholt** [cpp-netlib/cpp-netlib#552]
I have a strange issue when reusing existing response object. The code is like this:
```cxx
http::client::request req(some_path);
req << net::header(...);
http::client::response resp = m_client.get(req);
//now status(resp), headers(resp), resp.body() will block and return correct data
req = http::client::request(some_other_path);
req << net::header(...);
resp = m_client.get(req);
//now headers(resp), resp.body() return data from previous response! 
```

when I change the last line to this:
```cxx
http::client::response resp2 = m_client(req);
```
it will work.

What is the deal here? I would like to reuse same response object as I run requests in parallel and store responses in a single vector. Each iteration on the vector parses previous responses and sends new requests.

My environment:
cpp-netlib 0.11-1-final.zip
boost_1_57_0
VS2013
Win7 x64

**Comment by tpecholt** [comment]
OK so I found async_message class contains a lot more members than what is copied in operator= and constructor(const &) :( Especially retrieved_headers_ is used as a cache and because of faulty assignment the first successfully loaded state is kept. It's a dangerous code. operator= and copy constructor should be updated.

**Comment by deanberris** [comment]
Interesting. Can you see whether this has been fixed in 0.11-devel? If it isn't can you let me know and maybe also send a pull request to 0.11-devel to fix this? Thanks @tpecholt
